sys-boot/grub: update to 2.06 (#1082)
flatcar/scripts: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md

Flatcar uses a patched version of the GRUB, which implements the functionality to
read the [Flatcar Container Linux partition table](https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-disk-partitions/#partition-table)

## History

CoreOS Container Linux maintained a fork of the [grub](https://github.com/coreos/grub) and then was referenced
in the coreos-overlay. Any changes were made through [PRs](https://github.com/coreos/grub/pulls?q=is%3Apr+is%3Aclosed) to the grub repository.

> When Flatcar was born, a `grub` repo under the flatcar-linux namespace was created

Reviewer: I suppose the namespace is actually the GitHub org, right?

```suggestion
When Flatcar was born, a `grub` repo under the flatcar-linux org was created
```

> and referenced in the flatcar's coreos-overlay. Except for a few, now many changes

Reviewer: ```suggestion
and referenced in the Flatcar's coreos-overlay. Except for a few, now many changes
```

where brought into the system.

The repo was maintained at 2.02 version. During the 2.06 migration, the philosophy
to use a separate repo was scraped, and a single patch file was created. The patch

> file migrated only the essentials commits, and dropped all the other commits, which

Reviewer: Later it's mentioned that there are two patches, so let's use plural.

```suggestion
files migrated only the essential commits, and dropped all the other commits, which
```

> were either half-baked, or redundant at the point of migration. The two patch is applied

Reviewer: ```suggestion
were either half-baked, or redundant at the point of migration. The two patches are applied
```

on top of the grub sources, and emerge is done.

> Given below are the list of commits that were referenced to create the two patches

Reviewer: ```suggestion
Given below are the list of commits that were referenced to create the two patches.
```

## Summary of the patches

> The patch start with implementing a new implementation of reading the GPT instead

Reviewer: ```suggestion
The patch starts with adding a new implementation of reading the GPT instead
```

of using the traditional module. It provides essential functionality to interact
with GPT structures on disk, and checking/validating data integrity & GPT specification.

The commits goes on to add the following modules gptprio, gptrepair, and search
commands by label and partition.

The `gptprio` command which provides a mechanism to prioritize and select the

> next bootable partition based on the GPT attributed and results in flexible

Reviewer: ```suggestion
next bootable partition based on the GPT attributes and results in flexible
```

partition booting. The `gptrepair` command implements the repair functions for

> GPT information on a specified device. Few other functions include, searching

Reviewer: ```suggestion
GPT information on a specified device. Few other functions include searching
```

> devices by partition label, and partition UUID.

Reviewer: ```suggestion
devices by partition label or partition UUID.
```

## Commits

> Below are the commits that are picked to create the two patch for the grub. One is

Reviewer: ```suggestion
Below are the commits that are picked to create the two patches for the grub. One is
```
